ntbs-service/Pages/ContactDetails/Edit.cshtml.cs

```diff
@@ -45,8 +48,8 @@ public bool IsValid(string key)
 
         public async Task<IActionResult> OnPostAsync()
         {
-            var user = await _userService.GetUser(User);
-            if (ContactDetails.Username != user.Username)
+            var user = await _userRepository.GetUserByUsername(Username);
+            if (ContactDetails.Username != user.Username && !_userHelper.UserIsAdmin(HttpContext))
```

Comment:
This logic is essentially duplicated on the `ContactDetails` index page below - it would be good to commonise the two